WikidataGame/WikidataGame.Backend/Controllers/AuthController.cs

```diff
@@ -9,62 +9,83 @@
 using WikidataGame.Backend.Helpers;
 using WikidataGame.Backend.Repos;
 using WikidataGame.Backend.Services;
+using static WikidataGame.Backend.Repos.UserRepository;
 
 namespace WikidataGame.Backend.Controllers
 {
     [Route("api/[controller]")]
     [ApiController]
     public class AuthController : CustomControllerBase
     {
+        private readonly AuthService _authService;
+
         public AuthController(
             DataContext dataContext,
             IUserRepository userRepo,
             IGameRepository gameRepo,
             IRepository<Models.Category, string> categoryRepo,
-            INotificationService notificationService) : base(dataContext, userRepo, gameRepo, categoryRepo, notificationService)
-        {}
+            INotificationService notificationService,
+            AuthService authService) : base(dataContext, userRepo, gameRepo, categoryRepo, notificationService)
+        {
+            _authService = authService;
+        }
 
         /// <summary>
         /// Authenticates a player 
         /// </summary>
-        /// <param name="deviceId">device identifier</param>
+        /// <param name="username">Username</param>
+        /// <param name="authToken">firebase authentication token</param>
         /// <param name="pushToken">push token generated through firebase/apns</param>
         /// <returns>Information on how to authenticate</returns>
         [HttpGet]
         [ProducesResponseType(typeof(AuthInfo), StatusCodes.Status200OK)]
         public async Task<IActionResult> Authenticate(
-            [FromHeader(Name = "X-Device-ID")] string deviceId,
+            [FromHeader(Name = "X-Username")] string username,
+            [FromHeader(Name = "X-Auth-Token")] string authToken,
             [FromHeader(Name = "X-Push-Token")] string pushToken)
         {
-            if (string.IsNullOrWhiteSpace(deviceId))
-                return BadRequest(new { message = "DeviceId needs to be supplied" });
+            if (string.IsNullOrWhiteSpace(authToken))
```

Comment:
You don't check `string.IsNullOrWhiteSpace(username)` here, is that on purpose?

Reply:
Nicely spotted, I will check on that.